floor/test/integration/sqlparser_test/sqlparser_test.dart

```diff
@@ -0,0 +1,139 @@
+import 'dart:async';
+
+import 'package:floor/floor.dart';
+import 'package:floor_annotation/floor_annotation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sqflite/sqflite.dart' as sqflite;
+
+import '../model/person.dart';
+
+part 'sqlparser_test.g.dart';
+
+void main() {
+  TestDatabase database;
+  DeepDao deepDao;
+  CaseDao caseDao;
+
+  setUp(() async {
+    database = await $FloorTestDatabase.inMemoryDatabaseBuilder().build();
+    deepDao = database.deepDao;
+    caseDao = database.caseDao;
+  });
+
+  tearDown(() async {
+    await database.close();
+  });
+
+  test('Stream works with wrong case', () async {
+    final person1 = Person(1, 'Baba');
+    final person2 = Person(2, 'Me');
+
+    final actual = caseDao.allPersons();
+    expect(
+        actual,
+        emitsInOrder(<List<Person>>[
+          [],
+          [],
+          [person1],
+          [person1],
+          [person1, person2],
+          [person1, person2],
+        ]));
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.updateName();
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.insertPerson(person1);
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.updateName();
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.insertPerson(person2);
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.updateName();
+  });
+
+  group('Function type derivation tests', () {
+    test('try IN with null list', () async {
+      final list = ['a', 'b', 'c', 'd', 'e', 'f', null];
+
+      final aInList = await deepDao.isXinList('a', list);
+      expect(aInList, equals(true));
```

Comment:
❔ The SQLite database returns "a" in this case? Which use cases do you see for mapping a result (of not `INTEGER`) to true when the return type is `bool` ?

Reply:
In my case it returns `1` and not `"a"`. This is also what I want to test: a simple function that does not refer to any table and returns a primitive boolean (or null) value.
```
sqlite> SELECT "a" IN ("a", "2", "s");
1
```
`isXinList` is just a complicated way to write `['a', 'b', 'c', 'd', 'e', 'f', null].contains('a')`.